src/common/src/array/stream_chunk.rs

```diff
@@ -174,6 +174,43 @@ impl StreamChunk {
         StreamChunk::new(new_ops, columns)
     }
 
+    /// Split the `StreamChunk` into multiple chunks with the given size at most.
+    ///
+    /// When the total cardinality of all the chunks is not evenly divided by the `size`,
+    /// the last new chunk will be the remainder.
+    ///
+    /// For consecutive `UpdateDelete` and `UpdateInsert`, they will be kept in one chunk.
+    /// As a result, some chunks may have `size + 1` rows.
+    pub fn split(&self, size: usize) -> Vec<Self> {
+        let data_types = self.data_types();
+        let mut rows = Vec::with_capacity(size + 1);
+        let mut results = vec![];
+
+        let mut iter = self.rows();
+        while let Some(row) = iter.next() {
+            rows.push(row);
```

Comment:
We may leverage `StreamChunkBuilder` and `append_record` here.

- There's no need to allocate rows here and datums directly go to the new chunk.
- `UpdateDelete` is handled there.